test/errorHandling.js

```diff
@@ -132,19 +132,20 @@ describe('#errorHandling', () => {
         });
       });
 
-      it('should use errorHandler for an unresolvable host', done => {
-        // Test is not applicable for uds/stream transports
-        if (!['tcp', 'udp'].includes(serverType)) {
+      it('should use errorHandler for an unresolvable host with cacheDns', done => {
+        // this does not work for tcp/uds, which throws an error during setup
+        // that needs errorHandler or a socket.on('error') handler
+        if (serverType !== 'udp') {
           return done();
         }
+
         statsd = createHotShotsClient({
           host: '...',
+          cacheDns: true,
```

Comment:
This is the one important change in this file, the rest is cleanup.  Setting this exposed the bug that is now fixed